### PR TITLE
Update metering csv export format and intergrate metering module.

### PR DESCRIPTION
**Pull request by yunkunrao**
**What type of PR is this?**
/kind api-change

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for reviewers**:
```
```

**Additional documentation, usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apiserver/config/config.go [hunk 1]
```diff
@@ -207,6 +210,11 @@ func (conf *Config) ToMap() map[string]bool {
 			}
 		}
 
+		if name == "metering" {
+			result[name] = conf.MeteringOptions.Enable
+			continue
+		}
+
```
Comment (zheng1) [inline pkg/apiserver/config/config.go:217]:
```suggestion
```

Comment (zheng1) [reply pkg/apiserver/config/config.go:217]: Should add some code into `stripEmptyOptions`

**Review by zheng1 — CHANGES_REQUESTED** [review]

File: pkg/apiserver/config/config.go (hunk 1 above)
Comment (yunkunrao) [reply pkg/apiserver/config/config.go:217]: Got it

File: pkg/models/metering/type.go [hunk 2]
```diff
@@ -251,13 +315,28 @@ func (ds *DaemonsetStatistic) Aggregate() {
 }
 
 type ResourceStatistic struct {
-	Apps         map[string]*AppStatistic         `json:"apps" description:"app statistic"`
-	Services     map[string]*ServiceStatistic     `json:"services" description:"service statistic"`
+	// op statistic
+	Ops map[string]*OpStatistic `json:"ops" description:"op statistic"`
```
Comment (zheng1) [inline pkg/models/metering/type.go:319]: Should use the full name `openpitrix` instead of `op`

File: pkg/models/tenant/metering.go [hunk 3]
```diff
@@ -815,150 +748,186 @@ func (t *tenantOperator) listServices(user user.Info, ns string) (*corev1.Servic
...
+func (t *tenantOperator) isAppComponent(ns, kind, componentName string) (bool, string) {
 
-	ws := nsObj.Labels[constants.WorkspaceLabelKey]
+	appWorkloads := t.mo.GetAppWorkloads(ns, nil)
 
-	if len(ws) != 0 && ws != "system-workspace" {
-		return true
+	for k, cList := range appWorkloads {
+		for _, c := range cList {
+			if c == strings.Title(kind)+":"+componentName {
```
Comment (zheng1) [inline pkg/models/tenant/metering.go:875]:
```suggestion
			if c == fmt.Sprintf("%s:%s", strings.Title(kind), componentName) {
```

File: tools/cmd/doc-gen/main.go [hunk 4]
```diff
@@ -123,13 +126,13 @@ func generateSwaggerJson() []byte {
 	urlruntime.Must(devopsv1alpha2.AddToContainer(container, informerFactory.KubeSphereSharedInformerFactory(), &fakedevops.Devops{}, nil, clientsets.KubeSphere(), fakes3.NewFakeS3(), "", nil))
 	urlruntime.Must(devopsv1alpha3.AddToContainer(container, &fakedevops.Devops{}, clientsets.Kubernetes(), clientsets.KubeSphere(), informerFactory.KubeSphereSharedInformerFactory(), informerFactory.KubernetesSharedInformerFactory()))
 	urlruntime.Must(iamv1alpha2.AddToContainer(container, nil, nil, group.New(informerFactory, clientsets.KubeSphere(), clientsets.Kubernetes()), nil))
-	urlruntime.Must(monitoringv1alpha3.AddToContainer(container, clientsets.Kubernetes(), nil, nil, informerFactory))
+	urlruntime.Must(monitoringv1alpha3.AddToContainer(container, clientsets.Kubernetes(), nil, nil, informerFactory, ksClient))
```
Comment (zheng1) [inline tools/cmd/doc-gen/main.go:129]: Why not set nil as the input argument

File: pkg/models/metering/type.go (hunk 2 above)
Comment (yunkunrao) [reply pkg/models/metering/type.go:319]: got it

File: pkg/models/tenant/metering.go (hunk 3 above)
Comment (yunkunrao) [reply pkg/models/tenant/metering.go:875]: got it

File: tools/cmd/doc-gen/main.go (hunk 4 above)
Comment (yunkunrao) [reply tools/cmd/doc-gen/main.go:129]: got it

File: tools/cmd/doc-gen/main.go [hunk 5]
```diff
@@ -123,13 +126,13 @@ func generateSwaggerJson() []byte {
...
 	urlruntime.Must(iamv1alpha2.AddToContainer(container, nil, nil, group.New(informerFactory, clientsets.KubeSphere(), clientsets.Kubernetes()), nil))
-	urlruntime.Must(monitoringv1alpha3.AddToContainer(container, clientsets.Kubernetes(), nil, nil, informerFactory))
+	urlruntime.Must(monitoringv1alpha3.AddToContainer(container, clientsets.Kubernetes(), nil, nil, informerFactory, nil))
 	urlruntime.Must(openpitrixv1.AddToContainer(container, informerFactory, fake.NewSimpleClientset(), nil))
 	urlruntime.Must(openpitrixv2.AddToContainer(container, informerFactory, fake.NewSimpleClientset(), nil))
 	urlruntime.Must(operationsv1alpha2.AddToContainer(container, clientsets.Kubernetes()))
 	urlruntime.Must(resourcesv1alpha2.AddToContainer(container, clientsets.Kubernetes(), informerFactory, ""))
 	urlruntime.Must(resourcesv1alpha3.AddToContainer(container, informerFactory, nil))
-	urlruntime.Must(tenantv1alpha2.AddToContainer(container, informerFactory, nil, nil, nil, nil, nil, nil, nil, nil, nil))
+	urlruntime.Must(tenantv1alpha2.AddToContainer(container, informerFactory, nil, ksClient, nil, nil, nil, nil, nil, nil, nil))
```
Comment (zheng1) [inline tools/cmd/doc-gen/main.go:135]: remove ksClient

File: pkg/models/tenant/metering.go [hunk 6]
```diff
@@ -815,150 +748,186 @@ func (t *tenantOperator) listServices(user user.Info, ns string) (*corev1.Servic
...
+		if err != nil {
+			klog.Error(err)
+			return err
+		}
 
-			pods, err := t.listPods(user, ns, statefulset.Spec.Selector)
-			if err != nil {
-				klog.Error(err)
-				return err
+		if ok, _ := t.isOpComponent(cluster, ns, "statefulset", statefulset.Name); ok {
```
Comment (zheng1) [inline pkg/models/tenant/metering.go:898]: isOpComponent = > isOpenPitrixComponent

File: pkg/models/tenant/metering.go [hunk 7]
```diff
@@ -815,150 +748,186 @@ func (t *tenantOperator) listServices(user user.Info, ns string) (*corev1.Servic
...
-		return false
-	}
+func (t *tenantOperator) isAppComponent(ns, kind, componentName string) (bool, string) {
 
-	ws := nsObj.Labels[constants.WorkspaceLabelKey]
+	appWorkloads := t.mo.GetAppWorkloads(ns, nil)
 
-	if len(ws) != 0 && ws != "system-workspace" {
-		return true
+	for k, cList := range appWorkloads {
```
Comment (zheng1) [inline pkg/models/tenant/metering.go:873]: k => appName

Comment (zheng1) [inline pkg/models/tenant/metering.go:873]: k => appName

File: pkg/models/metering/type.go [hunk 8]
```diff
@@ -251,13 +315,28 @@ func (ds *DaemonsetStatistic) Aggregate() {
...
+	// app crd statistic
+	Apps map[string]*AppStatistic `json:"apps" description:"app statistic"`
+
+	// k8s workload only which exclude app and op
 	Deploys      map[string]*DeploymentStatistic  `json:"deployments" description:"deployment statistic"`
 	Statefulsets map[string]*StatefulsetStatistic `json:"statefulsets" description:"statefulset statistic"`
 	Daemonsets   map[string]*DaemonsetStatistic   `json:"daemonsets" description:"daemonsets statistics"`
 }
 
+func (rs *ResourceStatistic) GetOpStats(name string) *OpStatistic {
```
Comment (zheng1) [inline pkg/models/metering/type.go:330]: GetOpStats  => GetOpenPitrixStats

File: pkg/models/metering/type.go [hunk 9]
```diff
@@ -251,13 +315,28 @@ func (ds *DaemonsetStatistic) Aggregate() {
 }
 
 type ResourceStatistic struct {
-	Apps         map[string]*AppStatistic         `json:"apps" description:"app statistic"`
-	Services     map[string]*ServiceStatistic     `json:"services" description:"service statistic"`
+	// openpitrix statistic
+	Openpitrixs map[string]*OpStatistic `json:"ops" description:"op statistic"`
```
Comment (zheng1) [inline pkg/models/metering/type.go:319]: ops => openpitrixs

File: pkg/models/metering/type.go [hunk 10]
```diff
@@ -47,46 +52,105 @@ func (ps *PodsStats) Set(podName, meterName string, value float64) {
 	}
 }
 
+type OpStatistic struct {
```
Comment (zheng1) [inline pkg/models/metering/type.go:55]: OpStatistic => OpenPitrixStatistic

File: tools/cmd/doc-gen/main.go (hunk 5 above)
Comment (yunkunrao) [reply tools/cmd/doc-gen/main.go:135]: got it

File: pkg/models/tenant/metering.go (hunk 6 above)
Comment (yunkunrao) [reply pkg/models/tenant/metering.go:898]: got it

File: pkg/models/tenant/metering.go (hunk 7 above)
Comment (yunkunrao) [reply pkg/models/tenant/metering.go:873]: got it

Comment (yunkunrao) [reply pkg/models/tenant/metering.go:873]: got it

File: pkg/models/metering/type.go (hunk 8 above)
Comment (yunkunrao) [reply pkg/models/metering/type.go:330]: got it

File: pkg/models/metering/type.go (hunk 10 above)
Comment (yunkunrao) [reply pkg/models/metering/type.go:55]: got it

File: pkg/models/metering/type.go (hunk 9 above)
Comment (yunkunrao) [reply pkg/models/metering/type.go:319]: got it

**Review by zheng1 — APPROVED** [review]

**Comment by zryfish** [comment]
@yunkunrao Please squash commits, and run `hack/update-goimports.sh`

**Comment by yunkunrao** [comment]
> @yunkunrao Please squash commits, and run `hack/update-goimports.sh`

Got it

**Pull request merged**